controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -541,6 +780,58 @@ func (pp *portPublisher) endpointsToAddresses(endpoints *corev1.Endpoints) Addre
 	}
 }
 
+func (pp *portPublisher) newServiceRefAddress(endpointPort Port, endpointIP, serviceName, serviceNamespace string) (Address, ServiceID) {
+	id := ServiceID{
+		Name: strings.Join([]string{
+			serviceName,
+			endpointIP,
+			fmt.Sprint(endpointPort),
+		}, "-"),
+		Namespace: serviceNamespace,
+	}
+
+	return Address{IP: endpointIP, Port: endpointPort}, id
+}
+
+func (pp *portPublisher) newPodRefAddress(endpointPort Port, endpointIP, podName, podNamespace string) (Address, PodID, error) {
+	id := PodID{
+		Name:      podName,
+		Namespace: podNamespace,
+	}
+	pod, err := pp.k8sAPI.Pod().Lister().Pods(id.Namespace).Get(id.Name)
+	if err != nil {
+		return Address{}, PodID{}, fmt.Errorf("unable to fetch pod %v:%v", id, err)
+	}
+	ownerKind, ownerName := pp.k8sAPI.GetOwnerKindAndName(pod, false)
+	addr := Address{
+		IP:        endpointIP,
+		Port:      endpointPort,
+		Pod:       pod,
+		OwnerName: ownerName,
+		OwnerKind: ownerKind,
+	}
+
+	return addr, id, nil
+}
+
+func (pp *portPublisher) resolveESTargetPort(slicePorts []discovery.EndpointPort) Port {
+	if slicePorts == nil {
```

Comment:
WDYT about making that a bit more explicit. You can return a nil or an error instead of treating `Port(0)` as a special value. Is there any reason why you chose this approach?

Reply:
Nothing that I can think of other than I thought it might be consistent. When there isn't a defined requested port, we return `Port(0)`, so thought it would add the context we need. I think `nil` is a good idea though! Error might be a bit too much in this instance, I would rather return a nil since this function is used just once in the whole file when we translate the slice addresses.

Reply:
Yes I think simply returning nil makes sense here. Error is too much

Reply:
Because of the type though it will have to return `Port(0)`. The underlying data type for `Port` is `uint32`, so we can't have a nil return type, unfortunately.

Reply:
In a more sane programming language this would return an `Option[Port]`, but alas.  Using `Port(0)` as a special zero value that indicates that the port does not exist seems like a reasonably go-idiomatic thing to do.